src/prog_models/linear_model.py

```diff
@@ -64,25 +111,42 @@ def _propertyCheck(self, rowsCount : int, colsCount : int, notes : list) -> None
         target_property = getattr(self, notes[0])
         if isinstance(target_property, list):
             setattr(self, notes[0], np.array(target_property))
-        matrix = getattr(self, notes[0]) 
+        matrix = getattr(self, notes[0])
         if not isinstance(matrix, np.ndarray):
-            raise TypeError("Matrix type check failed: @property {} dimensions is not of type list or NumPy array.".format(notes[0]))
+            raise TypeError(
+                "Matrix type check failed: @property {} dimensions is not of type list or NumPy array.".format(notes[0]))
 
         matrixShape = matrix.shape
-        if (matrixShape[0] != rowsCount or # check matrix is 2 dimensional, correspond to rows count
-            len(matrixShape) == 1 or # check .shape returns 2-tuple, meaning all rows are of equal length
-            matrixShape[1] != colsCount or # check all rows are equal to correct column count
-            matrix.ndim != 2): # check matrix is 2 dimensional
-            raise AttributeError("Matrix size check failed: @property {} dimensions improperly formed along {} x {}.".format(notes[0],notes[1],notes[2]))
-   
+
+        if (matrix.ndim != 2 or  # Checks too see if matrix is two-dimensional
+            # checks if matrix has correct row count
+            matrixShape[0] != rowsCount or
+                matrixShape[1] != colsCount):  # check all rows are equal to correct column count
+            raise AttributeError(
+                "Matrix size check failed: @property {} dimensions improperly formed along {} x {}.".format(notes[0], notes[1], notes[2]))
+
     @property
     @abstractmethod
     def A(self):
         pass
 
     @property
     def B(self):
-        return np.zeros((self.n_states, self.n_inputs))
+        return self.parameters['_B']
+
+    @B.setter
+    def B(self, value):
+        if (value is None):
+            self.parameters['_B'] = np.zeros((self.n_states, self.n_inputs))
+        else:
+            prev_value = self.parameters['_B']
+            self.parameters['_B'] = value
+            try:
+                self._propertyCheck(self.n_states, self.n_inputs, [
+                                    "B", "outputs", "inputs"])
```

Comment:
From my understanding, yes, it is supposed to be self._propertyCheck(self.n_states, self.n_inputs, ["B", "states", "inputs"]).

How this does not fail the tests is a bit concerning, since I did not test a variety of different states and input lengths to see if anything breaks. 

Should I add that functionality to linear_test_models?

Reply:
I agree that it is a bit concerning - great thought! Anytime I find a bug, I always try to think about how we could make a test for that. Yes, let's dig into it a bit and see if we can write a test that would check this functionality.